src/services/oauth2/hooks/index.js

```diff
@@ -38,16 +38,24 @@ const setSubject = (hook) => {
 };
 
 const setIdToken = (hook) => {
+	const scope = hook.params.consentRequest.requested_scope;
 	if (!hook.params.query.accept) return hook;
 	return Promise.all([
 		hook.app.service('users').get(hook.params.account.userId),
+		scope.includes('groups')
+			? hook.app.service('teams').find({
```

Comment:
you can directly filter the fields of the result here, from the top of my head should be something like
`hook.app.service('teams').find({query: {'userIds.userId': hook.params.account.userId,},}).select({'name', 'id'})`